### ISSUE TITLE
[Suggestion] [2.0] Do not deprecate Vue.config.delimiters and Vue.config.unsafeDelimiters

### ISSUE THREAD

**Issue by funkatron** [vuejs/vue#2766]
I'd like to suggest that deprecating the delimiter config settings could be a mistake.

While I understand the motivation to suggest that one "consider switching to a better one", the choice of server-side templating engines may not up to the developer who made the choice for Vuejs. Given the 9-month "long-term support" schedule for security fixes, that may put some developers who have invested in VueJS when it turned 1.0 (like I did) in a bit of a pinch -- time to rewrite functional UI code is often hard to justify, on top of making arguments for a different templating engine.

If it is not an undue technical burden, I think support for changing delimiters should be retained.

Thanks!

**Comment by boussou** [comment]
+1

**Comment by blake-newman** [comment]
I agree, having to switch template systems to upgrade to 2.0, isn't great. Could turn out to be a lot of work, especially if that template solution is heavily used.

**Comment by prog-rajkamal** [comment]
Also, server side is not the only place of conflict. If I were to migrate from angular to vue then different delimiters can make a lot of difference.

**Comment by rpkilby** [comment]
I think there was a reason behind this. eg, a 3rd party component template uses `{{ }}`, but you've change the delimiters to `{% %}`. Would this not cause the 3rd party component to break?

Maybe it could be reworked as a Vue option instead of a global configuration? eg, 
```js
import Vue from 'vue';

const AppVue = Vue.extend({
    delimiters: '...',
});

export default AppVue;
```
Your app would import an extended Vue that's configured with your presets, or it could just manually set the delimiters per Vue.

**Comment by blake-newman** [comment]
@rpkilby This is what we ( @vuejs/collaborators ) were just discussing. We are currently thinking about enabling this locally to instances, rather than a global option. It's still in discussion, as we have to discover any possible side effects.

**Comment by yyx990803** [comment]
@rpkilby that is precisely one of the reasons for deprecation. Rework it as a component-level option makes sense as a compromise, and we'll look into that.

**Comment by funkatron** [comment]
That sounds great @yyx990803. Thanks!

**Comment by yyx990803** [comment]
This has been implemented in the 2.0 branch. Note that `unsafeDelimiters` is still deprecated - Vue 2.0 no longer supports injecting HTML at arbitrary positions - you will always need to use `v-html` on a containing element.
